src/Domain/Fund.php

```diff
@@ -75,4 +89,54 @@ public function getCurrencyCode(): string
     {
         return $this->currencyCode;
     }
+
+    /**
+     * Uses database copies of all data, not the Redis `Matching\Adapter`. Intended for use
+     * after a campaign closes.
+     *
+     * @return array{totalAmount: numeric-string, usedAmount: numeric-string, currencyCode: string}
+     */
+    public function getAmounts(): array
+    {
+        $totalAmount = Money::fromPoundsGBP(0);
+        $usedAmount = Money::fromPoundsGBP(0);
+
+        foreach ($this->campaignFundings as $campaignFunding) {
+            $thisAmount = Money::fromNumericStringGBP($campaignFunding->getAmount());
+            $thisAmountAvailable = Money::fromNumericStringGBP($campaignFunding->getAmountAvailable());
+            $thisAmountUsed = $thisAmount->minus($thisAmountAvailable);
+
+            $totalAmount = $totalAmount->plus($thisAmount);
+            $usedAmount = $usedAmount->plus($thisAmountUsed);
+        }
+
+        return [
+            'totalAmount' => $totalAmount->toNumericString(),
+            'usedAmount' => $usedAmount->toNumericString(),
+            'currencyCode' => $this->currencyCode,
+        ];
```

Comment:
IMHO better to delay calling toNumericString until as late as possible, so we have as little as possible stringly-typed. In this case that would mean calling it inside `toAmountUsedUpdateModel` not here. I think returning the money objects would make the test clearer and reduce the chance of mistakes in future calls to `getAmounts`

Reply:
So far `getAmounts()` is only used by the fn right after it and isn't tested on its own. I think I expected the 2 new fns to end up longer when I made them distinct. I'm now wondering if it makes more sense to merge them if no separate use cases some up so that we don't have to think about this distinction, what do you think?

Reply:
I think it's probably worth keeping this separate for testability, since its implementing some important and non-trivial domain logic.

Reply:
Thanks, I've aimed to do that in 77f77e5a9aa08dd0f55d3111c61bc234b22e2d56